libraries/api/src/middleware/queries.js

```diff
@@ -1,17 +1,22 @@
-import _ from 'lodash';
-
 /**
  * Validate query-string against query.
  *
  * Query-string options not specified in options will not be allowed. But it's
  * optional if a request carries any query-string parameters at all.
+ *
+ * @template {Record<string, any>} TContext
+ * @param {{
+ *   entry: import('../../@types/index.d.ts').APIEntryOptions<TContext>,
```

Comment:
If the imports become too cumbersome you could create a global namespace you import once through a `@typedef`, and then do something like `T.APIEntryOptions<TContext>`.

Reply:
Oh nice! I didn't know you could "import" whole file like this with typedef 👍  cool